src/main/drivers/timer_hal.c

```diff
@@ -801,89 +806,117 @@ _TIM_IRQ_HANDLER(TIM8_BRK_TIM12_IRQHandler, 12);
+    _CASE(3);
 #endif
 #if USED_TIMERS & TIM_N(4)
-    __HAL_RCC_TIM4_CLK_ENABLE();
+    _CASE(4);
 #endif
 #if USED_TIMERS & TIM_N(5)
-    __HAL_RCC_TIM5_CLK_ENABLE();
+    _CASE(5);
 #endif
 #if USED_TIMERS & TIM_N(6)
-    __HAL_RCC_TIM6_CLK_ENABLE();
+    _CASE(6);
 #endif
 #if USED_TIMERS & TIM_N(7)
-    __HAL_RCC_TIM7_CLK_ENABLE();
+    _CASE(7);
 #endif
 #if USED_TIMERS & TIM_N(8)
-    __HAL_RCC_TIM8_CLK_ENABLE();
+    _CASE(8);
 #endif
 #if USED_TIMERS & TIM_N(9)
-    __HAL_RCC_TIM9_CLK_ENABLE();
+    _CASE(9);
 #endif
 #if USED_TIMERS & TIM_N(10)
-    __HAL_RCC_TIM10_CLK_ENABLE();
+    _CASE(10);
 #endif
 #if USED_TIMERS & TIM_N(11)
-    __HAL_RCC_TIM11_CLK_ENABLE();
+    _CASE(11);
 #endif
 #if USED_TIMERS & TIM_N(12)
-    __HAL_RCC_TIM12_CLK_ENABLE();
+    _CASE(12);
 #endif
 #if USED_TIMERS & TIM_N(13)
-    __HAL_RCC_TIM13_CLK_ENABLE();
+    _CASE(13);
 #endif
 #if USED_TIMERS & TIM_N(14)
-    __HAL_RCC_TIM14_CLK_ENABLE();
+    _CASE(14);
 #endif
 #if USED_TIMERS & TIM_N(15)
-    __HAL_RCC_TIM15_CLK_ENABLE();
+    _CASE(15);
 #endif
 #if USED_TIMERS & TIM_N(16)
-    __HAL_RCC_TIM16_CLK_ENABLE();
+    _CASE(16);
 #endif
 #if USED_TIMERS & TIM_N(17)
-    __HAL_RCC_TIM17_CLK_ENABLE();
+    _CASE(17);
+#endif
+    default: return NULL;
+    }
+#undef _CASE
+}
 #endif
 
-    /* enable the timer peripherals */
-    for (int i = 0; i < USABLE_TIMER_CHANNEL_COUNT; i++) {
-        RCC_ClockCmd(timerRCC(timerHardware[i].tim), ENABLE);
+void timerInit(void)
+{
+    memset(timerConfig, 0, sizeof(timerConfig));
+
+#ifdef USE_TIMER_MGMT
+    for (unsigned i = 0; i < TIMER_CHANNEL_COUNT; i++) {
+        timerHardware_t *th = &timerHardware[i];
+        th->tim = getTimByTag(timerTags[i]);
+        th->tag = timerChannelConfig(i)->ioTag;
+        th->channel = CC_CHANNEL_FLAG_FROM_INDEX(TIMER_TAG_CHANNEL_INDEX(timerTags[i]));
+        th->output = timerChannelConfig(i)->inverted ? TIMER_OUTPUT_N_CHANNEL : TIMER_OUTPUT_NONE;
+        th->alternateFunction = timerAlternateFunction(timerTags[i], timerChannelConfig(i)->ioTag);
+        th->dmaRef = dmaGetRefByIdentifier(timerChannelConfig(i)->dmaIdentifier);
+#if defined(STM32F7) || defined(STM32F4)
+        th->dmaChannel = timerChannelConfig(i)->dmaChannel;
+#endif
+        th->dmaIrqHandler = timerChannelConfig(i)->dmaIdentifier;
     }
 
+#else
 #if defined(STM32F3) || defined(STM32F4) || defined(STM32F7)
-    for (int timerIndex = 0; timerIndex < USABLE_TIMER_CHANNEL_COUNT; timerIndex++) {
+    for (unsigned timerIndex = 0; timerIndex < USABLE_TIMER_CHANNEL_COUNT; timerIndex++) {
         const timerHardware_t *timerHardwarePtr = &timerHardware[timerIndex];
         if (timerHardwarePtr->usageFlags == TIM_USE_NONE) {
             continue;
         }
         // XXX IOConfigGPIOAF in timerInit should eventually go away.
         IOConfigGPIOAF(IOGetByTag(timerHardwarePtr->tag), IOCFG_AF_PP, timerHardwarePtr->alternateFunction);
     }
+#endif
 #endif
 
+    /* enable the timer peripherals */
```

Comment:
First initialization should enable timer ... Enabling them here may mask some types of bugs in code

Reply:
I would argue first use of the IO pin should do the same. I will leave for another PR as it is existing functionality.